src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.Fdc3.DesktopAgent/Fdc3StartupAction.cs

```diff
@@ -43,26 +44,36 @@ public Fdc3StartupAction(
-            try
-            {
-                var appId = (await _appDirectory.GetApp(startupContext.StartRequest.ModuleId)).AppId;
-                var userChannelSet = await _userChannelSetReader.GetUserChannelSet();
-                var fdc3InstanceId = startupContext.StartRequest.Parameters.FirstOrDefault(parameter => parameter.Key == Fdc3StartupParameters.Fdc3InstanceId).Value ?? Guid.NewGuid().ToString();
-                
-                //TODO: decide if we want to join to a channel automatically on startup of an fdc3 module
-                //First we inject the channel id if we set as startup parameter eg.: when we open an app with `fdc3.open`
-                var channelId = startupContext.StartRequest.Parameters.FirstOrDefault(parameter => parameter.Key == Fdc3StartupParameters.Fdc3ChannelId).Value ?? _options.ChannelId ?? userChannelSet.FirstOrDefault().Key;
-                
-                var openedAppContextId =
-                    startupContext.StartRequest.Parameters.FirstOrDefault(
-                        x => x.Key == Fdc3StartupParameters.OpenedAppContextId).Value;
-
-                var fdc3StartupProperties = new Fdc3StartupProperties() { InstanceId = fdc3InstanceId, ChannelId = channelId, OpenedAppContextId = openedAppContextId };
-                fdc3InstanceId = startupContext.GetOrAddProperty<Fdc3StartupProperties>(_ => fdc3StartupProperties).InstanceId;
+            var appId = (await _appDirectory.GetApp(startupContext.StartRequest.ModuleId)).AppId;
+            var userChannelSet = await _userChannelSetReader.GetUserChannelSet();
+
+            var fdc3InstanceId = startupContext
+                .StartRequest
+                .Parameters
+                .FirstOrDefault(parameter => parameter.Key == Fdc3StartupParameters.Fdc3InstanceId).Value
+                                 ?? Guid.NewGuid().ToString();
+
+            //TODO: decide if we want to join to a channel automatically on startup of an fdc3 module
+            //First we inject the channel id if we set as startup parameter eg.: when we open an app with `fdc3.open`
+            var channelId = startupContext
+                .StartRequest
+                .Parameters
+                .FirstOrDefault(parameter => parameter.Key == Fdc3StartupParameters.Fdc3ChannelId).Value ?? _options.ChannelId
+                    ?? userChannelSet.FirstOrDefault().Key;
+
+            var openedAppContextId = startupContext
+                .StartRequest
+                .Parameters
+                .FirstOrDefault(x => x.Key == Fdc3StartupParameters.OpenedAppContextId).Value;
+
+            var fdc3StartupProperties = new Fdc3StartupProperties { InstanceId = fdc3InstanceId, ChannelId = channelId, OpenedAppContextId = openedAppContextId };
+            fdc3InstanceId = startupContext.GetOrAddProperty<Fdc3StartupProperties>(_ => fdc3StartupProperties).InstanceId;
 
+            if (startupContext.ModuleInstance.Manifest.ModuleType == ModuleType.Web)
```

Comment:
Consider creating an interface or abstract class to replace this conditional that executes the steps inside depending on implementation